hudi-common/src/main/java/org/apache/hudi/hadoop/CachingPath.java

```diff
@@ -32,11 +33,12 @@
  * NOTE: This class is thread-safe
  */
 @ThreadSafe
-public class CachingPath extends Path implements Serializable {
```

Comment:
`Path` is not serializable right? Have we vetted all paths that this does not get sent to executors (to avoid TaskNotSerializable execption)